nullaway/src/test/java/com/uber/nullaway/tools/ErrorDisplay.java

```diff
@@ -50,16 +77,19 @@ public boolean equals(Object o) {
     }
     ErrorDisplay that = (ErrorDisplay) o;
     return type.equals(that.type)
-        // To increase readability, a shorter version of the actual message might be present in the
-        // expected output of tests.
+        && encMethod.equals(that.encMethod)
+        && encClass.equals(that.encClass)
+        && kind.equals(that.kind)
+        && clazz.equals(that.clazz)
         && (message.contains(that.message) || that.message.contains(message))
-        && method.equals(that.method)
-        && clazz.equals(that.clazz);
+        && variable.equals(that.variable)
+        && index.equals(that.index)
+        && uri.equals(that.uri);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(type, message, method, clazz);
+    return Objects.hash(type, message, encMethod, encClass);
```

Comment:
Now that this has a lot more fields, shouldn't those be part of its hashCode()? This was all fields before, no?

Reply:
This was all fields before, but according to the usage of these classes which are mostly used in test cases and are compared to at most `3/4` other instances, I did not see it necessary to include all fields. But you are right, it is better to be more precise about the hash.

04b3664d6a8f121e96dfb3547b7f8785f75f4b5a